pkg/compare/compare.go

```diff
@@ -437,6 +456,20 @@ func extractPath(str string, pathIndex int) string {
 	return "Unknown Path"
 }
 
+func diffAgainstTemplate(temp *ReferenceTemplate, clusterCR *unstructured.Unstructured, o *Options) (*bytes.Buffer, error) {
+	localRef, err := temp.Exec(clusterCR.Object)
+	if err != nil {
+		return nil, err
+	}
+	obj := InfoObject{
+		injectedObjFromTemplate: localRef,
+		clusterObj:              clusterCR,
+		FieldsToOmit:            temp.FieldsToOmit(o.ref.FieldsToOmit),
+		allowMerge:              temp.Config.AllowMerge,
+	}
+	return runDiff(obj, o.IOStreams, o.ShowManagedFields)
```

Comment:
Does it make sense to have `runDiff` as a separate function? This is the only time its used from what I can tell.

Reply:
missed that, removed it :)